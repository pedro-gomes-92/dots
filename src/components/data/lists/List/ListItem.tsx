import React, { FunctionComponent } from 'react';
import { BaseListItemProps, BaseListItem } from '../BaseListItem';

export interface ListItemProps extends BaseListItemProps {}

export const ListItem: FunctionComponent<ListItemProps> = ({ children, ...rest }: ListItemProps): JSX.Element => (
  <BaseListItem {...rest}>{children}</BaseListItem>
);
