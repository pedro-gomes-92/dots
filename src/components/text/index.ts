export * from './Text';
export * from './Label';
export * from './Paragraph';
export * from './Title';
export * from './Tag';
