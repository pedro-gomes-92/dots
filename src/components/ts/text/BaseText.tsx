import React, { FunctionComponent } from 'react';
import classnames from 'classnames';
import { BaseProps } from '../Base';
import { Base } from '../Base';

export interface BaseTextProps extends BaseProps {
  text: string;
  isBold?: boolean;
  size?: FontSize | Size;
  align?: AlignXY;
}

export const BaseText: FunctionComponent<BaseTextProps> = ({
  className,
  text,
  align,
  isBold,
  ...rest
}: BaseTextProps): JSX.Element => (
  <Base
    className={classnames(
      'text',
      align && align.horizontal ? `has-alignment-horizontal-${align.horizontal}` : '',
      align && align.vertical ? `has-alignment-vertical-${align.vertical}` : '',
      { 'is-bold': isBold },
      className,
    )}
    {...rest}
  >
    {text}
  </Base>
);

BaseText.defaultProps = {
  isBold: false,
};
