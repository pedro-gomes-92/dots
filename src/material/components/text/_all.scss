@import 'text';
@import 'title';
@import 'label';
@import 'paragraph';
@import 'tag';
