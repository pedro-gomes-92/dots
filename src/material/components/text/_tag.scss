/*
 * tag.scss
 *
 * All styling applied to all tag components
 *
 * Index
 * - Tag
 */
.tag {
  text-transform: uppercase;
}
