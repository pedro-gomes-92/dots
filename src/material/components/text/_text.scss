/*
 * text.scss
 *
 * All styling applied to all text components
 *
 * Index
 * - Text
 *    - Sizes
 *    - Bold
 *    - Color
 */
.text {
  white-space: pre-wrap;

  &.is {
    // Sizes
    @each $name, $size in $text-sizes {
      &-#{$name} {
        font-size: $size;
      }
    }

    &-bold {
      // Bold
      font-weight: bold;
    }

    @each $name, $pair in $colors {
      // Color
      &-#{$name} {
        color: nth($pair, 1);
      }
    }
  }
}
