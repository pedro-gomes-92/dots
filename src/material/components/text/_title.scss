/*
 * title.scss
 *
 * All styling applied to the title component
 *
 * Index
 * - Title
 */

.title {
  font-family: $text-primary-family;
}
