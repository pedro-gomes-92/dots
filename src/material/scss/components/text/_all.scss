@import 'text';
@import 'label';
@import 'paragraph';
