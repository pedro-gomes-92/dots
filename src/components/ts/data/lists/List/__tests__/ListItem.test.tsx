import React from 'react';
import { render } from '@testing-library/react';

import { Text } from '../../../../text';
import { ListItem } from '../ListItem';

describe('Given <ListItem />', (): void => {
  let container: HTMLElement;

  describe('When initializes', (): void => {
    it('Then should match the snapshot', (): void => {
      container = render(
        <ListItem>
          <Text text="Testing" />
        </ListItem>,
      ).container;
      expect(container).toMatchSnapshot();
    });
  });
});
