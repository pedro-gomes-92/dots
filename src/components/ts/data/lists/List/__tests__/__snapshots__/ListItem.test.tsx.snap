// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <ListItem /> When initializes Then should match the snapshot 1`] = `
<div>
  <li
    class="list-item"
  >
    <span
      class="text"
    >
      Testing
    </span>
  </li>
</div>
`;
